Fix 1

Starworm opened the pull request:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + maps enum

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors

Kuorti commented on src/app/components/country-selector/country-selector.component.html:
А зачем отдельный компонент для выбора страны? Вроде библиотечный селект должен подойти

Starworm replied:
исправил на стандартный

Kuorti commented on src/app/components/country-selector/country-selector.component.html:
```diff
@@ -1,8 +1,9 @@
-        <option *ngFor="let country of countryList"
-                [value]="country.lat + ',' + country.lon + ',' + country.zoom + ',' + country.id">{{country.name}}</option>
+        @for (country of countryList; track country.id) {
+            <option [value]="country.latitude + ',' + country.longitude + ',' + country.zoom + ',' + country.id">{{country.name}}</option>
```
Если этот компонент останется, то вместо `country.latitude + ',' + country.longitude + ',' + country.zoom + ',' + country.id` нужна пайпа типа [value]="country | asGeographicalCoords". Имя условно написал, потому что не знаю, для чего это конкретно

Starworm replied:
не останется, убрал за ненадобностью

Kuorti commented on src/app/components/login/login.component.html:
```diff
@@ -3,28 +3,32 @@
-                            <div *ngIf="loginForm.get('username')?.invalid && loginForm.get('username')?.touched" class="text-danger">
-                                Username is required.
-                            </div>
+                            @if (!loginForm.controls['username'].valid && loginForm.controls['username'].touched) {
```
```suggestion
                            @if (!loginForm.controls.username.valid && loginForm.controls.username.touched) {
```

Starworm replied:
это ж контрол, у него могут быть динамические свойства, лучше оставить через скобки

Kuorti commented on src/app/components/login/login.component.html:
```diff
@@ -3,28 +3,32 @@
-                            </div>
+                            @if (!loginForm.controls['password'].valid && loginForm.controls['password'].touched) {
+                                <div class="text-danger">
+                                    Password is required and must be at least 6 characters.
```
Здесь и в остальных местах нужно выносить в словарь

Starworm replied:
исправлено

Kuorti commented on src/app/components/login/login.component.ts:
```diff
@@ -42,11 +45,14 @@ export class LoginComponent {
      */
-    public forgetPassword() {
+    toResetPasswordPage() {
         this.router.navigate(['resetpassword']);
```
Роуты нужно вынести в константы в отдельный файл в папке модуля. Типа файл app.routes.ts, в нём
`
export const APP_ROUTES = {
resetPassword: 'resetpassword', ...
}
`

Starworm replied:
исправлено

Kuorti commented on src/app/components/map/map.component.ts:
```diff
@@ -67,73 +76,36 @@ export class MapComponent implements OnInit {
-        // this.markerClusterGroup.current?.remove();
+    zoomToCountry(coords: CountryInterface, zoomLevel: number = 5) {
+        console.log(coords);
+        this.map.setView([coords.latitude, coords.longitude], coords.zoom ? coords.zoom : zoomLevel);
```
```suggestion
        this.map.setView([coords.latitude, coords.longitude], coords.zoom ?? zoomLevel);
```

Starworm replied:
исправил

Kuorti commented on src/app/components/map/map.component.ts:
```diff
@@ -67,73 +76,36 @@ export class MapComponent implements OnInit {
+            .pipe(
+                takeUntilDestroyed(this.destroyRef)
+            )
             .subscribe((res) => {
```
Здесь можно через map
this.markerService.getData(coords.id).pipe(map((markers: ТИП): GeojsonObjectInterface => markers.features))
`

Kuorti replied:
Судя по коду запрос данных происходит каждый раз при выхове zoomToCoutry. И каждый раз данные добавляются в массивы concertGroup, boardgamesGroup. Выглядит как баг

Kuorti replied:
Вообще не оч понятно для чего этот метод нужен. Вроде его данные не используются. Если он всё таки нужен, то его нужно распилить на отдельные методы. Данные по концертам и играм сделать обзёрвабалами.
То есть должно быть что-то вроде
`this.concertGroup$ = this.markerService.getData().pipe(switchMap((data) => this.createConcertsFromMarkerData(data)))`
А в шаблоне использовать их через async пайпу
`<article>{{ concertGroup$ | async }}</article>`

Kuorti replied:
Также вопрос зачем делить так жёстко в коде на категории. Это будет сложно поддерживать при необходимости добавления новой категории мест, скажем, мастерклассов

Kuorti replied:
Внутри методов нужно стараться не использовать обращение к глобальным переменным, это приводит к сложноотлавливаемым багам, когда одновременно одно и то же поле перетирается в разных местах

Starworm replied:
> Здесь можно через map this.markerService.getData(coords.id).pipe(map((markers: ТИП): GeojsonObjectInterface => markers.features)) `

можно. Но через подписку выглядит понятнее

Starworm replied:
> Судя по коду запрос данных происходит каждый раз при выхове zoomToCoutry. И каждый раз данные добавляются в массивы concertGroup, boardgamesGroup. Выглядит как баг

да, я про это говорил, надо править. Так же, как разобраться с кластерами. Но вначале надо определится, переходим мы на маплибру или остаемся на лифлете

Starworm replied:
> Вообще не оч понятно для чего этот метод нужен. Вроде его данные не используются. Если он всё таки нужен, то его нужно распилить на отдельные методы. Данные по концертам и играм сделать обзёрвабалами. То есть должно быть что-то вроде `this.concertGroup$ = this.markerService.getData().pipe(switchMap((data) => this.createConcertsFromMarkerData(data)))` А в шаблоне использовать их через async пайпу `<article>{{ concertGroup$ | async }}</article>`

ну как же не используются? Они добавляются в список мероприятий, которые показываются на карте при переключении на страну. В перспективе можно наверное на что-то изменить, особенно если данные по мероприятиям будут приходить по каждому отдельно

Starworm replied:
> Также вопрос зачем делить так жёстко в коде на категории. Это будет сложно поддерживать при необходимости добавления новой категории мест, скажем, мастерклассов

это для фильтрации надо. Иначе на карту будут одновременно вываливаться все события, что может быть неудобно. Как альтернатива, можно подумать о хэштэгах, которые использовать вместо категорий

Kuorti commented on src/app/components/map/map.component.ts:
```diff
@@ -167,6 +139,35 @@ export class MapComponent implements OnInit {
+
+
+        // provider of map (currently OpenStreetMap)
+        const tiles = L.tileLayer(this.MAP_TEMPLATE, {
```
Здесь нужно применить паттерн стратегии, чтобы не привязываться к конкретной либе
https://refactoring.guru/ru/design-patterns/strategy

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors

Commit:
Fixing:
  + eslint-angular
  - .spec files
  - 'public' access modifiers
  - 'any' types
  + titles enum
  +- loginForm.get(...) -> loginForm.controls....
  +- form validation
  + enums
  + unsubscribing in component methods
  + new syntax @if and @for
  - country enum
  +- properties names
  +- service providing
  - empty constructors
